hcn/hcnnetwork.go

```diff
@@ -453,15 +453,17 @@ func (network *HostComputeNetwork) Create() (*HostComputeNetwork, error) {
 		for _, subnet := range ipam.Subnets {
 			if subnet.IpAddressPrefix != "" {
 				hasDefault := false
+				needsDefault := false
```

Comment:
>false [](http://example.com/codeflow?start=20&length=5)

decide the value of needsDefault by checking if subnet.Routes is empty before the loop, no need to set it in the loop.